### PR TITLE
[FLINK-17291][docs] Translate 'docs/training/event_driven.zh.md' to C…

### PR DESCRIPTION
**Pull request by RocMarshal**
Translate 'docs/training/event_driven.zh.md' to Chinese

## What is the purpose of the change

*Rich Chinese documents of Flink*

## Brief change log

  - Translate 'docs/training/event_driven.zh.md' to Chinese 

## Verifying this change

  - Translate 'docs/training/event_driven.zh.md' to Chinese 
  
## Does this pull request potentially affect one of the following parts:

  - docs/training/event_driven.zh.md

## Documentation

  - This pull request doesn't  introduce a new feature

### CODE REVIEW

File: docs/training/event_driven.zh.md [hunk 1]
```diff
@@ -29,19 +29,18 @@ under the License.
...
+### 介绍
 
-A `ProcessFunction` combines event processing with timers and state, making it a powerful building
-block for stream processing applications. This is the basis for creating event-driven applications
-with Flink. It is very similar to a `RichFlatMapFunction`, but with the addition of timers.
+`ProcessFunction` 将事件处理与 Timer 和 State 结合在一起，使其成为流处理应用的强大构建模块。
+这是使用 Flink 创建事件驱动的应用程序的基础。它和 `RichFlatMapFunction` 十分相似， 但是增加了 Timer。
 
-### Example
+### 实例
```
Comment (XBaith) [inline docs/training/event_driven.zh.md:37]: Do you mean “示例” instead of “实例”？

File: docs/training/event_driven.zh.md [hunk 2]
```diff
@@ -29,19 +29,18 @@ under the License.
...
 
-If you've done the
-[hands-on exercise]({% link training/streaming_analytics.zh.md %}#hands-on)
-in the [Streaming Analytics training]({% link training/streaming_analytics.zh.md %}),
-you will recall that it uses a `TumblingEventTimeWindow` to compute the sum of the tips for
-each driver during each hour, like this:
+如果你已经体验了
+[流式分析训练]({% link training/streaming_analytics.zh.md %})
+的[动手实践]({% link training/streaming_analytics.zh.md %}#hands-on)，
+你会忆起，它是采用 `TumblingEventTimeWindow` 来计算每个小时内每个司机的小费总和，
```
Comment (XBaith) [inline docs/training/event_driven.zh.md:42]: “忆起”还是说“想起”，我个人觉得“想起”读起来更符合我们平常的说法。亦或者说意译成“你应该记得”

File: docs/training/event_driven.zh.md [hunk 3]
```diff
@@ -51,8 +50,8 @@ DataStream<Tuple3<Long, Long, Float>> hourlyTips = fares
         .process(new AddTips());
 {% endhighlight %}
 
-It is reasonably straightforward, and educational, to do the same thing with a
-`KeyedProcessFunction`. Let us begin by replacing the code above with this:
+使用 `KeyedProcessFunction` 去实现相同的效果是合理、直接且有学习意义的。
```
Comment (XBaith) [inline docs/training/event_driven.zh.md:53]:
```suggestion
使用 `KeyedProcessFunction` 去实现相同的操作更加直接且更有学习意义。
```

File: docs/training/event_driven.zh.md [hunk 4]
```diff
@@ -202,52 +196,48 @@ public void onTimer(
...
 
 {% top %}
 
 ## Side Outputs
 
-### Introduction
+### 介绍
 
-There are several good reasons to want to have more than one output stream from a Flink operator, such as reporting:
+有几个很好的理由希望从 Flink operator 获得多个输出流，如下报告条目：
```
Comment (XBaith) [inline docs/training/event_driven.zh.md:221]:
```suggestion
有几个很好的理由希望从 Flink 算子获得多个输出流，如下报告条目：
```

File: docs/training/event_driven.zh.md [hunk 5]
```diff
@@ -269,38 +259,35 @@ SingleOutputStreamOperator hourlyTips = fares
...
-[Rides and Fares Exercise](https://github.com/apache/flink-training/tree/{% if site.is_stable %}release-{{ site.version_title }}{% else %}master{% endif %}/rides-and-fares),
-where a `RichCoFlatMapFunction` is used to compute a simple join, the sample solution assumes that
-the TaxiRides and TaxiFares are perfectly matched, one-to-one for each `rideId`. If an event is lost,
-the other event for the same `rideId` will be held in state forever. This could instead be implemented
-as a `KeyedCoProcessFunction`, and a timer could be used to detect and clear any stale state.
+ProcessFunctions 的另一个常见用例是过期过时 State。如果你回想一下
+[Rides and Fares Exercise](https://github.com/apache/flink-training/tree/{% if site.is_stable %}release-{{ site.version_title }}{% else %}master{% endif %}/rides-and-fares)，
+其中使用 `RichCoFlatMapFunction` 来计算简单 Join，那么示例解决方案假设 TaxiRides 和 TaxiFares 
+完全匹配，每个 `rideId` 一对一。如果某个事件丢失，则同一 `rideId` 的另一个事件将永远保持 State。
+这可以作为 `Keyedcomprocessfunction` 实现，并且可以使用计时器来检测和清除任何过时的 State。
```
Comment (XBaith) [inline docs/training/event_driven.zh.md:279]:
```suggestion
这可以作为 `KeyedCoProcessFunction` 实现，并且可以使用计时器来检测和清除任何过时的 State。
```

Comment (RocMarshal) [reply docs/training/event_driven.zh.md:279]: There is no doubt that it was a mistake of proofreading before I submitted the document. I will correct and put an end to such mistakes in time.
Thank you very much !

File: docs/training/event_driven.zh.md (hunk 1 above)
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: Yes, in order to be consistent, your suggestion is feasible. I have made changes. Thank you very much for your attention.

File: docs/training/event_driven.zh.md (hunk 2 above)
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: Yes, it will be easier to understand in translation. I've updated it.
thank you.

**Comment by RocMarshal** [comment]
@flinkbot run travis

**Comment by RocMarshal** [comment]
Hi, @XBaith ,I have updated the document according to your suggestions. Obviously, it is very helpful for the translation of the document. Thank you very much for your help.

**Comment by RocMarshal** [comment]
Hi, @wuchong . I have completed the translation of this page and made corresponding improvements according to the suggestions of community members. If you have free time, would you please review it for me?

Thank you  very  much.

**Comment by RocMarshal** [comment]
Hi, XBaith .
I have completed the translation of this page and made corresponding improvements according to the your suggestions . If you have free time, would you please review it for me?

Thank you very much.

Best,
Roc



在 2020-05-04 01:25:04，"Xu Bai" <notifications@github.com> 写道：

@XBaith commented on this pull request.

Good job!
I leave a few suggestions that may could help.

In docs/training/event_driven.zh.md:

>  
-### Example
+### 实例


Do you mean “示例” instead of “实例”？

In docs/training/event_driven.zh.md:

>  
-If you've done the
-[hands-on exercise]({% link training/streaming_analytics.zh.md %}#hands-on)
-in the [Streaming Analytics training]({% link training/streaming_analytics.zh.md %}),
-you will recall that it uses a `TumblingEventTimeWindow` to compute the sum of the tips for
-each driver during each hour, like this:
+如果你已经体验了
+[流式分析训练]({% link training/streaming_analytics.zh.md %})
+的[动手实践]({% link training/streaming_analytics.zh.md %}#hands-on)，
+你会忆起，它是采用 `TumblingEventTimeWindow` 来计算每个小时内每个司机的小费总和，


“忆起”还是说“想起”，我个人觉得“想起”读起来更符合我们平常的说法。亦或者说意译成“你应该记得”

In docs/training/event_driven.zh.md:

> @@ -51,8 +50,8 @@ DataStream<Tuple3<Long, Long, Float>> hourlyTips = fares
         .process(new AddTips());
 {% endhighlight %}
 
-It is reasonably straightforward, and educational, to do the same thing with a
-`KeyedProcessFunction`. Let us begin by replacing the code above with this:
+使用 `KeyedProcessFunction` 去实现相同的效果是合理、直接且有学习意义的。

⬇️ Suggested change
-使用 `KeyedProcessFunction` 去实现相同的效果是合理、直接且有学习意义的。
+使用 `KeyedProcessFunction` 去实现相同的操作更加直接且更有学习意义。


In docs/training/event_driven.zh.md:

>  
-There are several good reasons to want to have more than one output stream from a Flink operator, such as reporting:
+有几个很好的理由希望从 Flink operator 获得多个输出流，如下报告条目：

⬇️ Suggested change
-有几个很好的理由希望从 Flink operator 获得多个输出流，如下报告条目：
+有几个很好的理由希望从 Flink 算子获得多个输出流，如下报告条目：


In docs/training/event_driven.zh.md:

>  
-Another common use case for ProcessFunctions is for expiring stale state. If you think back to the
-[Rides and Fares Exercise](https://github.com/apache/flink-training/tree/{% if site.is_stable %}release-{{ site.version_title }}{% else %}master{% endif %}/rides-and-fares),
-where a `RichCoFlatMapFunction` is used to compute a simple join, the sample solution assumes that
-the TaxiRides and TaxiFares are perfectly matched, one-to-one for each `rideId`. If an event is lost,
-the other event for the same `rideId` will be held in state forever. This could instead be implemented
-as a `KeyedCoProcessFunction`, and a timer could be used to detect and clear any stale state.
+ProcessFunctions 的另一个常见用例是过期过时 State。如果你回想一下
+[Rides and Fares Exercise](https://github.com/apache/flink-training/tree/{% if site.is_stable %}release-{{ site.version_title }}{% else %}master{% endif %}/rides-and-fares)，
+其中使用 `RichCoFlatMapFunction` 来计算简单 Join，那么示例解决方案假设 TaxiRides 和 TaxiFares 
+完全匹配，每个 `rideId` 一对一。如果某个事件丢失，则同一 `rideId` 的另一个事件将永远保持 State。
+这可以作为 `Keyedcomprocessfunction` 实现，并且可以使用计时器来检测和清除任何过时的 State。

⬇️ Suggested change
-这可以作为 `Keyedcomprocessfunction` 实现，并且可以使用计时器来检测和清除任何过时的 State。
+这可以作为 `KeyedCoProcessFunction` 实现，并且可以使用计时器来检测和清除任何过时的 State。

**Comment by RocMarshal** [comment]
@klion26 
Hi, @klion26 .
I have completed the translation of this page and made corresponding improvements according to the suggestions of community members. If you have free time, would you please review it for me?

Thank you for your attention.

**Comment by klion26** [comment]
@RocMarshal thanks for the contribution, I left some comments.

**Comment by RocMarshal** [comment]
> @RocMarshal thanks for the contribution, I left some comments.

@klion26   Thank you for your comments and help !
I'm a starter , what should I do next to promote this PR?

**Comment by RocMarshal** [comment]
@klion26 Thank you for your comments and help ! And I'm a starter.
Could you please tell me what should I do next to promote this PR (https://github.com/apache/flink/pull/11979)?

Thank you so much.




Best,

Roc




At 2020-05-06 16:09:16, "Congxian Qiu" <notifications@github.com> wrote:

@RocMarshal thanks for the contribution, I left some comments.

**Comment by RocMarshal** [comment]
@flinkbot run travis

**Comment by klion26** [comment]
@RocMarshal sorry for the inconvenience for my last reply, I just did not submit my comments. and I just submitted my comments, please take a look.

File: docs/training/event_driven.zh.md [hunk 6]
```diff
@@ -29,19 +29,18 @@ under the License.
 
 ## Process Functions
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md:30]: @klion26  What about translating ' Process Functions ' it into '处理函数' or '流程函数'？

File: docs/training/event_driven.zh.md [hunk 7]
```diff
@@ -29,19 +29,18 @@ under the License.
...
 
-### Introduction
+### 介绍
 
-A `ProcessFunction` combines event processing with timers and state, making it a powerful building
-block for stream processing applications. This is the basis for creating event-driven applications
-with Flink. It is very similar to a `RichFlatMapFunction`, but with the addition of timers.
+`ProcessFunction` 将事件处理与 Timer 和 State 结合在一起，使其成为流处理应用的强大构建模块。
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md:34]: @klion26 Thanks for your suggestion. I will make improvements based on your suggestions in next commit.

**Comment by RocMarshal** [comment]
> ## CI report:
> * [51f00c0](https://github.com/apache/flink/commit/51f00c041b126b54616f191f669750e819c733db) Azure: [PENDING](https://dev.azure.com/apache-flink/98463496-1af2-4620-8eab-a2ecc1a2e6fe/_build/results?buildId=543)
> * [e34fdb3](https://github.com/apache/flink/commit/e34fdb3a37d8c673e9349936f7a390cda5b87ce3) UNKNOWN
> * [c36ba53](https://github.com/apache/flink/commit/c36ba5313969ea13c44cced005de17ec88c82696) Azure: [PENDING](https://dev.azure.com/apache-flink/98463496-1af2-4620-8eab-a2ecc1a2e6fe/_build/results?buildId=547)
> * [0e69259](https://github.com/apache/flink/commit/0e6925923af7e86c8e4356fee25460b610040e69) UNKNOWN
> * [a7681f6](https://github.com/apache/flink/commit/a7681f61b126d12e9ab6ea62d0cdb4829bb342c2) Azure: [FAILURE](https://dev.azure.com/apache-flink/98463496-1af2-4620-8eab-a2ecc1a2e6fe/_build/results?buildId=752)
> * [0cc5466](https://github.com/apache/flink/commit/0cc54660550d76fe482a266f08afc504f4ed8f82) UNKNOWN
> 
> Bot commands
> The @flinkbot bot supports the following commands:
> * `@flinkbot run travis` re-run the last Travis build
> * `@flinkbot run azure` re-run the last Azure build

@flinkbot run azure

File: docs/training/event_driven.zh.md [hunk 8]
```diff
@@ -269,38 +259,35 @@ SingleOutputStreamOperator hourlyTips = fares
...
-Another common use case for ProcessFunctions is for expiring stale state. If you think back to the
-[Rides and Fares Exercise](https://github.com/apache/flink-training/tree/{% if site.is_stable %}release-{{ site.version_title }}{% else %}master{% endif %}/rides-and-fares),
-where a `RichCoFlatMapFunction` is used to compute a simple join, the sample solution assumes that
-the TaxiRides and TaxiFares are perfectly matched, one-to-one for each `rideId`. If an event is lost,
-the other event for the same `rideId` will be held in state forever. This could instead be implemented
-as a `KeyedCoProcessFunction`, and a timer could be used to detect and clear any stale state.
+ProcessFunctions 的另一个常见用例是过期过时 State。如果你回想一下
+[Rides and Fares Exercise](https://github.com/apache/flink-training/tree/{% if site.is_stable %}release-{{ site.version_title }}{% else %}master{% endif %}/rides-and-fares)，
+其中使用 `RichCoFlatMapFunction` 来计算简单 Join，那么示例解决方案假设 TaxiRides 和 TaxiFares 
+完全匹配，每个 `rideId` 一对一。如果某个事件丢失，则同一 `rideId` 的另一个事件将永远保持 State。
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: @klion26 
Hi, @klion26 
I re-integrated the entire scene of the example to translate these sentences more semantically and apply your suggestions. If you have any better suggestions, please leave  comments, I will continue to improve it. From the bottom of my heart, thank you so much for your guidance and help !

Sincerely,
Roc

File: docs/training/event_driven.zh.md [hunk 9]
```diff
@@ -269,38 +259,35 @@ SingleOutputStreamOperator hourlyTips = fares
...
+此外，`ProcessFunctions` 对于计算分析之外的许多其他用例也很有用。
+下面的实践练习提供了一个完全不同的例子。
 
-Another common use case for ProcessFunctions is for expiring stale state. If you think back to the
-[Rides and Fares Exercise](https://github.com/apache/flink-training/tree/{% if site.is_stable %}release-{{ site.version_title }}{% else %}master{% endif %}/rides-and-fares),
-where a `RichCoFlatMapFunction` is used to compute a simple join, the sample solution assumes that
-the TaxiRides and TaxiFares are perfectly matched, one-to-one for each `rideId`. If an event is lost,
-the other event for the same `rideId` will be held in state forever. This could instead be implemented
-as a `KeyedCoProcessFunction`, and a timer could be used to detect and clear any stale state.
+ProcessFunctions 的另一个常见用例是过期过时 State。如果你回想一下
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: @klion26 
Yes, it's more accurate! I have made corresponding improvements according your suggestion. Thank you so much.

Sincerely,
Roc

File: docs/training/event_driven.zh.md [hunk 10]
```diff
@@ -202,52 +196,48 @@ public void onTimer(
...
-### Example
+### 示例
 
-You are now in a position to do something with the late events that were ignored in the previous
-section.
+现在你可以对上一节中忽略的延迟事件执行某些操作。
 
-A side output channel is associated with an `OutputTag<T>`. These tags have generic types that
-correspond to the type of the side output's `DataStream`, and they have names.
+Side output channel 与 `OutputTag<T>` 相关联。这些标记具有与 Side output 的 `DataStream` 类型相对应的泛型类型，并且它们具有名称。
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: I have made corresponding improvements according your suggestion and the original text. Thank you so much.

Sincerely,
Roc

File: docs/training/event_driven.zh.md [hunk 11]
```diff
@@ -202,52 +196,48 @@ public void onTimer(
...
+  这相当于在使用 Flink 的时间窗口时将 allowedLateness 设置为零。
 
-### Performance Considerations
+### 性能考虑
 
-Flink provides `MapState` and `ListState` types that are optimized for RocksDB. Where possible,
-these should be used instead of a `ValueState` object holding some sort of collection. The RocksDB
-state backend can append to `ListState` without going through (de)serialization, and for `MapState`, each
-key/value pair is a separate RocksDB object, so `MapState` can be efficiently accessed and updated.
+在可能的情况下，应该使用这些对象，而不是持有某种集合的 `ValueState` 对象。
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: Hi, @klion26 
I have made some improvements according your suggestion and the original text. Thank 
 you.

Sincerely,
Roc

File: docs/training/event_driven.zh.md [hunk 12]
```diff
@@ -202,52 +196,48 @@ public void onTimer(
 }
 {% endhighlight %}
 
-Observations:
+观察：
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: Hi, @klion26 

I have replaced it with '注意：'. And  what do you think of it ?

Sincerely,
Roc

File: docs/training/event_driven.zh.md [hunk 13]
```diff
@@ -61,11 +60,10 @@ DataStream<Tuple3<Long, Long, Float>> hourlyTips = fares
...
-stream, the result of which is a `DataStream<Tuple3<Long, Long, Float>>` (the same kind of stream
-produced by the implementation that uses Flink's built-in time windows).
+在这个代码片段中，一个名为 `PseudoWindow` 的 `KeyedProcessFunction` 被应用于 KeyedStream ，
+其结果是一个 `DataStream<Tuple3<Long, Long, Float>>` （与使用 Flink 内置时间窗口的实现生成的流相同）。
 
-The overall outline of `PseudoWindow` has this shape:
+`PseudoWindow` 的总体轮廓示意如下：
 
 {% highlight java %}
 // Compute the sum of the tips for each driver in hour-long windows.
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: @klion26 
Hi , @klion26 
I have followed up the translation of the comments and I'm looking forward to receiving your suggestions
Thank you for your attention.

Sincerely,
Roc

File: docs/training/event_driven.zh.md [hunk 14]
```diff
@@ -106,21 +104,20 @@ public static class PseudoWindow extends
...
-  and `getRuntimeContext` methods needed for working with managed keyed state.
+* `KeyedProcessFunction` 是一种 `RichFunction`。作为 `RichFunction`，它可以访问使用 Managed Keyed State 所需的 `open`
+  和 `getRuntimeContext` 方法。
 
-* There are two callbacks to implement: `processElement` and `onTimer`. `processElement` is called
-  with each incoming event; `onTimer` is called when timers fire. These can be either event time or
-  processing time timers. Both `processElement` and `onTimer` are provided with a context object
-  that can be used to interact with a `TimerService` (among other things). Both callbacks are also
-  passed a `Collector` that can be used to emit results.
+* 有两个回调方法须要实现： `processElement` 和 `onTimer`。`processElement` 与每个传入事件一起调用；
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: Yes. @klion26 ,I handled it.
Thank you so much .

File: docs/training/event_driven.zh.md [hunk 15]
```diff
@@ -106,21 +104,20 @@ public static class PseudoWindow extends
...
+  当计时器触发时调用 `onTimer`。它们可以是事件时间（event time）timer，也可以是处理时间（processing time）timer。
+  `processElement` 和 `onTimer` 都提供了一个上下文对象，该对象可用于与 `TimerService` 交互（除其他外）。
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: Hi, @klion26 
What do you think of translating  
"`processElement` 和 `onTimer` 都提供了一个上下文对象，该对象可用于与 `TimerService` 交互（除其他外）。"         
into
"此外/除此之外，`processElement` 和 `onTimer` 都提供了一个上下文对象，该对象可用于与 `TimerService` 交互。" ？
Thank you for your attention!

Sincerely,
Roc

File: docs/training/event_driven.zh.md [hunk 16]
```diff
@@ -106,21 +104,20 @@ public static class PseudoWindow extends
...
+* `KeyedProcessFunction` 是一种 `RichFunction`。作为 `RichFunction`，它可以访问使用 Managed Keyed State 所需的 `open`
+  和 `getRuntimeContext` 方法。
 
-* There are two callbacks to implement: `processElement` and `onTimer`. `processElement` is called
-  with each incoming event; `onTimer` is called when timers fire. These can be either event time or
-  processing time timers. Both `processElement` and `onTimer` are provided with a context object
-  that can be used to interact with a `TimerService` (among other things). Both callbacks are also
-  passed a `Collector` that can be used to emit results.
+* 有两个回调方法须要实现： `processElement` 和 `onTimer`。`processElement` 与每个传入事件一起调用；
+  当计时器触发时调用 `onTimer`。它们可以是事件时间（event time）timer，也可以是处理时间（processing time）timer。
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: @klion26 

I have applied your suggestion into the translation of the sentence.

Sincerely,
Roc

File: docs/training/event_driven.zh.md [hunk 17]
```diff
@@ -106,21 +104,20 @@ public static class PseudoWindow extends
...
+  `processElement` 和 `onTimer` 都提供了一个上下文对象，该对象可用于与 `TimerService` 交互（除其他外）。
+  这两个回调还传递了一个可用于发出结果的 `Collector`。
 
-#### The `open()` method
+#### `open()` 方法
 
 {% highlight java %}
 // Keyed, managed state, with an entry for each window, keyed by the window's end time.
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: @klion26 
I have translated the comments into Chinese.

Sincerely,
Roc

File: docs/training/event_driven.zh.md [hunk 18]
```diff
@@ -136,13 +133,12 @@ public void open(Configuration conf) {
 }
 {% endhighlight %}
 
-Because the fare events can arrive out of order, it will sometimes be necessary to process events
-for one hour before having finished computing the results for the previous hour. In fact, if the
-watermarking delay is much longer than the window length, then there may be many windows open
-simultaneously, rather than just two. This implementation supports this by using a `MapState` that
-maps the timestamp for the end of each window to the sum of the tips for that window.
+由于票价事件（fare-event）可能会乱序到达，有时需要在计算完前一小时的结果之前处理一小时的事件。
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: @klion26 
I have followed up the translation of the comments and I'm looking forward to receiving your suggestions
Thank you for your attention.

Sincerely,
Roc

File: docs/training/event_driven.zh.md [hunk 19]
```diff
@@ -136,13 +133,12 @@ public void open(Configuration conf) {
...
+实际上，如果 Watermark 延迟比窗口长度长得多，则可能有多个窗口同时打开，而不仅仅是两个。
+此实现通过使用 `MapState` 来支持处理这一点，该 `MapState` 将每个窗口的结束时间戳映射到该窗口的 tips
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply docs/training/event_driven.zh.md]: @klion26 
I have translated the 'tips' into Chinese.

Sincerely,
Roc

**Comment by klion26** [comment]
@RocMarshal thanks for addressing my comments, LGTM now
ping @wuchong

**Pull request merged**